connector/src/main/scala/com/datastax/spark/connector/datasource/CassandraInJoinReaderFactory.scala

```diff
@@ -27,33 +28,32 @@ case class CassandraInJoinReaderFactory(
 
   override def createReader(partition: InputPartition): PartitionReader[InternalRow] =
     if (cqlQueryParts.selectedColumnRefs.contains(RowCountRef)) {
-      CassandraInJoinCountReader(connector, tableDef, inClauses, readConf, schema, cqlQueryParts, partition)
+      CassandraInJoinCountReader(connector, tableMetadata, inClauses, readConf, schema, cqlQueryParts, partition)
     } else {
-      CassandraInJoinReader(connector, tableDef, inClauses, readConf, schema, cqlQueryParts, partition)
+      CassandraInJoinReader(connector, tableMetadata, inClauses, readConf, schema, cqlQueryParts, partition)
     }
 }
 
 
-abstract class CassandraBaseInJoinReader(
-  connector: CassandraConnector,
-  tableDef: TableDef,
-  inClauses: Seq[In],
-  readConf: ReadConf,
-  schema: StructType,
-  cqlQueryParts: ScanHelper.CqlQueryParts,
-  partition: InputPartition)
+abstract class CassandraBaseInJoinReader(connector: CassandraConnector,
+                                         tableMetadata: TableMetadata,
+                                         inClauses: Seq[In],
+                                         readConf: ReadConf,
+                                         schema: StructType,
+                                         cqlQueryParts: ScanHelper.CqlQueryParts,
+                                         partition: InputPartition)
```

Comment:
IntelliJ kept insisting on reformatting arg lists in this way and nothing I did could convince it to stop. This is a relatively minor thing that can be cleaned up later but my apologies for the noise while reviewing.